apis/v1alpha1/cidr_collection.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CidrCollectionSpec defines the desired state of CidrCollection.
+//
+// A complex type that identifies a CIDR collection.
+type CIDRCollectionSpec struct {
+
+	// Information about changes to a CIDR collection.
+	// +kubebuilder:validation:Required
+	Locations []*CIDRCollectionChange `json:"locations"`
```

Comment:
Do we want this to be a `[][]CIDRBlocks` ? Also wondering if we really need the `Action` field.

Reply:
I agree, we technically do not need the Action field in the resource definition, however I used the generator to create this field, let me check if I can simplify this part as indeed it would be nicer to have a cidrBlock representation here.

Reply:
After investigation I do not think that a CIdrBlock type is the best here, there is just a string in it that contains all cidrblocks, not a list of strings. Furthermore I would have to complicate things in the genrator file to get that working. I would prefer to leave it like this.
Sure the Action field is not required as I set it in the code, but i could not manage to have it removed with the generator.